From a6647abfb02ed0ff37c72867166a1727cc79e42a Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Fri, 25 Jan 2019 10:30:15 +0800
Subject: [PATCH] SELinux: grant some permissions for vendor_init

Some kernel modules need to be loaded by vendor_init, grant
the relevant permissions to vendor_init.

Tracked-On: None

Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 sepolicy/domain.te      | 1 +
 sepolicy/vendor_init.te | 3 +++
 2 files changed, 4 insertions(+)
 create mode 100644 sepolicy/domain.te
 create mode 100644 sepolicy/vendor_init.te

diff --git a/sepolicy/domain.te b/sepolicy/domain.te
new file mode 100644
index 0000000..b256345
--- /dev/null
+++ b/sepolicy/domain.te
@@ -0,0 +1 @@
+allow domain system_file:dir r_dir_perms;
diff --git a/sepolicy/vendor_init.te b/sepolicy/vendor_init.te
new file mode 100644
index 0000000..aee5f7b
--- /dev/null
+++ b/sepolicy/vendor_init.te
@@ -0,0 +1,3 @@
+allow vendor_init self:global_capability_class_set sys_module;
+allow vendor_init vendor_file:system module_load;
+allow vendor_init kernel:key search;
-- 
2.20.1

